internal/controller/controller.go

```diff
@@ -223,36 +223,52 @@ func (c *Controller) startInformers(stopCh <-chan struct{}) {
 }
 
 func (c *Controller) createMeshServices() error {
-	services, err := c.clients.GetServices(metav1.NamespaceAll)
+	potentialSvcs, err := c.clients.GetServicesWithSelectors(
+		metav1.NamespaceAll,
+		c.ignored.LabelSelector(),
+		c.ignored.FieldSelector(),
+	)
 	if err != nil {
 		return fmt.Errorf("unable to get services: %v", err)
 	}
 
-	for _, service := range services {
-		if c.ignored.IsIgnoredService(service.Name, service.Namespace) {
-			continue
-		}
+	maeshSvcs, err := c.clients.GetServicesWithSelectors(c.meshNamespace, "app==maesh", "")
+	if err != nil {
+		return fmt.Errorf("unable to get maesh services: %v", err)
+	}
 
+	maeshSvcsByName := reduceServicesByName(maeshSvcs)
+
+	for _, service := range potentialSvcs {
 		log.Debugf("Creating mesh for service: %v", service.Name)
+
 		meshServiceName := c.userServiceToMeshServiceName(service.Name, service.Namespace)
 
-		for _, subservice := range services {
-			// If there is already a mesh service created, don't bother recreating
-			if subservice.Name == meshServiceName && subservice.Namespace == c.meshNamespace {
-				continue
-			}
+		if _, found := maeshSvcsByName[meshServiceName]; found {
+			log.Debugf("Skipping already created maesh service: %s", service.Name)
+			continue
 		}
 
 		log.Infof("Creating associated mesh service: %s", meshServiceName)
 
 		if err := c.createMeshService(service); err != nil {
-			return fmt.Errorf("unable to get create mesh service: %v", err)
+			return fmt.Errorf("unable to create mesh service: %v", err)
 		}
 	}
 
 	return nil
 }
 
+func reduceServicesByName(svcs []*corev1.Service) map[string]*corev1.Service {
```

Comment:
Could we rename this `makeServiceNameMap()` or `createServiceMapByName()` or something like that?

Reply:
There is a problem here, what if two services have the same name across two different namespaces ?

Reply:
what if the map key is `<name>/<namespace>`?

Reply:
or `map["name"]["namespace"]*corev1.Service`

Reply:
then we can reference the service directly

Reply:
Or I can use the service lister here, and not really have to build another index, WDYT ?

Reply:
Yes. Do it!

Reply:
> There is a problem here, what if two services have the same name across two different namespaces ?

Actually I'm working with services only from the maesh namespace service to it's fine to do this.